decidim-blogs/lib/decidim/api/blogs_type.rb

```diff
@@ -18,7 +18,7 @@ class BlogsType < Decidim::Api::Types::BaseObject
       end
 
       def posts(filter: {}, order: {})
-        Decidim::Core::ComponentListBase.new(model_class: Post).call(object, { filter:, order: }, context)
+        Decidim::Core::ComponentListBase.new(model_class: Post).call(object, { filter:, order: }, context).published
```

Comment:
We have the same problem here as with the single post preview.

All posts should be visible for logged in admin users, even when they are not published. They should be only hidden from the API for non-admin users and for visitors (i.e. non-logged in users).

We have currently similar functionality for participatory spaces (e.g. processes):
https://github.com/decidim/decidim/blob/9356312df607538c3ae7c1b3fc9eb85f373a3cfb/decidim-core/lib/decidim/api/functions/participatory_space_list_base.rb#L30

If you are an admin user, you will see all the processes through the API, even those that are unpublished. If you are a visitor or non-admin you can only see the processes that are published and visible for you.